client/apis/volumegroupsnapshot/v1alpha1/types.go

```diff
@@ -0,0 +1,284 @@
+	// The snapshot controller will keep retrying when an error occurs during the
+	// group snapshot creation. Upon success, this error field will be cleared.
+	// +optional
+	Error *snapshotv1.VolumeSnapshotError `json:"error,omitempty" protobuf:"bytes,4,opt,name=error,casttype=VolumeSnapshotError"`
+
+	// VolumeSnapshotRefList is the list of volume snapshot references for this
+	// group snapshot.
+	// The maximum number of allowed snapshots in the group is 100.
+	// +optional
+	VolumeSnapshotRefList []core_v1.ObjectReference `json:"volumeSnapshotRefList,omitempty" protobuf:"bytes,5,opt,name=volumeSnapshotRefList"`
+}
+
+//+genclient
+//+k8s:deepcopy-gen=true
+//+k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+//+kubebuilder:resource:scope=Namespaced,shortName=vgs
+//+kubebuilder:subresource:status
+
+// VolumeGroupSnapshot is a user's request for creating either a point-in-time
+// group snapshot or binding to a pre-existing group snapshot.
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+type VolumeGroupSnapshot struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	Spec   VolumeGroupSnapshotSpec   `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+	Status VolumeGroupSnapshotStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// VolumeGroupSnapshotList contains a list of VolumeGroupSnapshot
+type VolumeGroupSnapshotList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []VolumeGroupSnapshot `json:"items"`
+}
+
+//+genclient
+//+genclient:nonNamespaced
+//+k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VolumeGroupSnapshotClass is the Schema for the volumegroupsnapshotclass API
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,shortName=vsgclass;vsgclasses
+type VolumeGroupSnapshotClass struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// Driver is the name of the storage driver expected to handle this VolumeGroupSnapshotClass.
+	// Required.
+	Driver string `json:"driver" protobuf:"bytes,2,opt,name=driver"`
+
+	// Parameters is a key-value map with storage driver specific parameters for
+	// creating group snapshots.
+	// These values are opaque to the system and are passed directly
+	// to the driver.
+	// +optional
+	Parameters map[string]string `json:"parameters,omitempty" protobuf:"bytes,3,rep,name=parameters"`
+
+	// DeletionPolicy determines whether a VolumeGroupSnapshotContent created
+	// through the VolumeGroupSnapshotClass should be deleted when its bound
+	// VolumeGroupSnapshot is deleted.
+	// Supported values are "Retain" and "Delete".
```

Comment:
Add this:
// "Retain" means that the VolumeGroupSnapshotContent and its physical group snapshot on underlying storage system are kept.
	// "Delete" means that the VolumeGroupSnapshotContent and its physical group snapshot on underlying storage system are deleted.